### DISCUSSION TITLE
PlexCleaner conversion Options FFMPEG and Handbrake

### DISCUSSION THREAD

**Discussion by C0nw0nk** [ptr727/PlexCleaner#362] [General]
So its not a bug just to be clear thats why i created a discussion.

So with conversion options what also repairs broken media by default it seems PlexCleaner will opt to use FFMPEG then fallback to HandBrake.

Example.
https://github.com/ptr727/PlexCleaner/blob/main/PlexCleaner.defaults.json#L151

So here is my example logs

FFMPEG trys to repair first by default 

```
16:39:16 [INF] <1> Attempting media repair by ReEncoding using FfMpeg : "test.mkv"
16:39:16 [INF] <1> Executing FfMpeg : "-analyzeduration 2147483647 -probesize 2147483647 -hwaccel cuda -hwaccel_output_format cuda -i \"C:\\Users\\Videos\\New folder\\test.mkv\" -max_muxing_queue_size 1024 -abort_on empty_output -map 0 -c:v hevc_nvenc -crf 27 -preset veryslow -c:a ac3 -c:s copy -f matroska \"C:\\Users\\Videos\\New folder\\test.tmp\""
```

I intentionally made FFMPEG Fail because i use this line in my config.

```
"FfMpegOptions": {
      "Video": "hevc_nvenc -crf 27 -preset veryslow",
```

This will cause FFMPEG to fail with this error.

```
[hevc_nvenc @ 00000229360b00c0] [Eval @ 0000003d7dffe030] Undefined constant or missing '(' in 'veryslow'
```

The reason i do this is so that i can use handbrake to repair media since i do not want to use ffmpeg.

```
Conversion failed!
16:39:17 [ERR] <1> ReEncoding using FfMpeg failed : "test.mkv"
16:39:17 [INF] <1> Attempting media repair by ReEncoding using HandBrake : "test.mkv"
16:39:17 [INF] <1> Executing HandBrake :
```

Is there a configurable option to have only Handbrake function for conversions and repairs instead of ffmpeg or perhaps out of preference handbrake over ffmpeg so to change the order of how plexcleaner chooses to attempt repairs.

Its a bit of a hacky way to force plexcleaner to use handbrake for this instead of ffmpeg just curious if there was a easier way.

**Comment by ptr727** [comment]
No such option, in my experience ffmpeg is more resilient which is why it is tried first, and in my experience hb is better at deinterlacing. Why do you not want to use ffmpeg?

**Reply by C0nw0nk** [reply]
I would not say want because i would like to have ffmpeg as the second backup option just i prefer handbrake over ffmpeg since it tends to not fail so much for me as ffmpeg will fail with converting a corrupt media container and then handbrake would just convert it for me and not complain.

Just a difference in my prefrences to yours i guess.

I do use ffmpeg still i just like having it the other way around with ffmpeg as my fall back / go to option.
